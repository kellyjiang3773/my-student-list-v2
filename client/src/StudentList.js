import React from 'react';
import PropTypes from 'prop-types';
import Student from './Student';

const StudentList = (props) => {
    const studentNodes = props.data.map(student => (
        <li>
            <Student
                key={student._id}
                id={student._id}
                // handleDeleteStudent={props.handleDeleteStudent}
                // handleUpdateStudent={props.handleUpdateStudent}
            >
                {student.name}
            </Student>
        </li>
    ));
    return (
        <ul>
            {studentNodes}
        </ul>
    );
};

StudentList.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape(
        {
            name: PropTypes.string,
            id: PropTypes.string,
            // aMark: PropTypes.string,
            // mMark: PropTypes.string,
            // fMark: PropTypes.string
        }
    )),
    // handleDeleteStudent: PropTypes.func,
    // handleUpdateStudent: PropTypes.func,
};

StudentList.defaultProps = {
    data: [],
};

export default StudentList;