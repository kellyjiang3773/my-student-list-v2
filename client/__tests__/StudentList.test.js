import StudentList from '../src/components/StudentList';
import React from 'react';
// import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('DOM test for <StudentList>', () => {

    const mockValues = [
        {name: 'Isaac Newton', _id: '1', url: '1'},
        {name: 'Albert Einstein', _id: '2', url: '2'},
        {name: 'Tycho Brahe', _id: '3', url: '3'}
    ];

    let rendered;
    const setup = (data) => {
        rendered = shallow(
            <StudentList
                data={data}
            />
        );
    };

    it('renders a list of students', () => {
        setup(mockValues);
        expect(toJson(rendered)).toMatchSnapshot();
    });

    it('renders a message when list is empty', () => {
        setup([]);
        expect(rendered.find('li').first().text()).toEqual('There are no students');
    });

    it('link points to right direction', () => {
        setup(mockValues);
        expect(rendered.find('Link').first().props().to).toBe(mockValues[0].url);
        expect(rendered.find('Link').at(1).props().to).toBe(mockValues[1].url);
        expect(rendered.find('Link').at(2).props().to).toBe(mockValues[2].url);
    });
});