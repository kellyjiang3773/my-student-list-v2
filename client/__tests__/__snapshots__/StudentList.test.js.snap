// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM test for <StudentList> renders a list of students 1`] = `
<ul>
  <li
    key="1"
  >
    <Link
      replace={false}
      to="1"
    >
      Isaac Newton
    </Link>
  </li>
  <li
    key="2"
  >
    <Link
      replace={false}
      to="2"
    >
      Albert Einstein
    </Link>
  </li>
  <li
    key="3"
  >
    <Link
      replace={false}
      to="3"
    >
      Tycho Brahe
    </Link>
  </li>
</ul>
`;
